README.md

```diff
@@ -46,11 +47,12 @@ Install Manually
 
 ```php
 <?php
-// import Mixpanel
-require 'mixpanel-php/lib/Mixpanel.php';
 
-// get the Mixpanel class instance, replace with your project token
-$mp = Mixpanel::getInstance("MIXPANEL_PROJECT_TOKEN");
+use Mixpanel\Factory\MixpanelFactory;
+
+// get the Mixpanel class instance from the Mixpanel factory, replace with your project token
+$factory = new MixpanelFactory("MIXPANEL_PROJECT_TOKEN");
```

Comment:
@jbwyme - this must be one of the oldest replies to an open github PR! Anyway, having recently come back to look at it, I'm not sure the singleton way seems best. The problem I see with it is that testing becomes difficult. In the example given above, getInstance() is a static public function of Mixpanel (i.e. Mixpanel class is not an instance). I therefore can't change the dependencies of the Mixpanel class dynamically (i.e. through injection). I can't therefore inject mock classes to test different parts of the functionality. This also makes it harder for 3rd party developers who almost certainly won't want to call the mixpanel api in their unit/functional tests and will need to write their own mock mixpanel classes to avoid the api calls. This is the case for me at the moment, writing a mixpanel symfony bundle. We could get around it with something like a MockMixpanel class but we'd need as many of these as there are different permutations of the class - for instance, MockMixpanelWithFactoryA, MockMixpanelWithFactoryB etc. I guess it becomes a question of how important the BC is / whether another branch is suitable etc? Open to other suggestions too! :-)